status/views.py

```diff
@@ -0,0 +1,107 @@
+  settings are missing or invalid, to prevent information leakage.
+* Different services provide different information, but all should return
+  "up," "down", or "no config found" for the "status" key.
+"""
+
+import re
+from datetime import datetime
+import logging
+
+from django.conf import settings
+from django.http import JsonResponse
+from elasticsearch import Elasticsearch, ConnectionError as ESConnectionError
+from psycopg2 import connect, OperationalError
+from redis import StrictRedis, ConnectionError as RedisConnectionError
+
+log = logging.getLogger(__name__)
+
+
+def get_pg_info():
+    """Check PostgreSQL connection."""
+    try:
+        conf = settings.DATABASES['default']
+    except (AttributeError, KeyError):
+        log.error("No PostgreSQL connection info found in settings.")
+        return {"status": "no config found"}
+    try:
+        start = datetime.now()
+        connect(
+            database=conf["NAME"], user=conf["USER"], host=conf["HOST"],
+            port=conf["PORT"], password=conf["PASSWORD"]
+        )
+        micro = (datetime.now() - start).microseconds
+    except (OperationalError, KeyError):
+        log.error("Invalid PostgreSQL connection info in settings: %s", conf)
+        return {"status": "down"}
+    return {"status": "up", "response_ms": micro}
+
+
+def get_redis_info():
+    """Check Redis connection."""
+    pat = re.compile(r'redis://([^:]+):(\d+)/(\d+)$')
+    try:
+        url = settings.BROKER_URL
+    except AttributeError:
+        log.error("No Redis connection info found in settings.")
+        return {"status": "no config found"}
+    try:
+        match = pat.match(url)
+    except TypeError:
+        match = None
+    if match is None:
+        log.error("Invalid Redis connection info found in settings: %s", url)
+        return {"status": "no config found"}
+    host, port, database = match.groups()
+    rdb = StrictRedis(host=host, port=port, db=database)
+    start = datetime.now()
+    try:
+        info = rdb.info()
+    except RedisConnectionError:
+        return {"status": "down"}
+    micro = (datetime.now() - start).microseconds
+    ret = {
+        "status": "up", "response_ms": micro,
+    }
+    fields = ("uptime_in_seconds", "used_memory", "used_memory_peak")
+    ret.update({x: info[x] for x in fields})
+    return ret
+
+
+def get_elasticsearch_info():
+    """Check Elasticsearch connection."""
+    pat = re.compile(r'([^:]+):(\d+)$')
+    try:
+        url = settings.HAYSTACK_CONNECTIONS["default"]["URL"]
+    except (AttributeError, KeyError):
+        log.error("No Elasticsearch connection info in settings.")
+        return {"status": "no config found"}
+    match = pat.match(url)
+    if match is None:
+        log.error("Invalid Elasticsearch connection info: %s.", url)
+        return {"status": "no config found"}
+    host, port = match.groups()
+    search = Elasticsearch(hosts=[{"host": host, "port": port}])
```

Comment:
Elasticsearch can just take the string as it is in haystack so you don't need to parse it at all.  However, I would check the `HAYSTACK_CONNECTIONS` default and make sure it is of the ElasticSearch type above before testing, so that we could change it to solr for example, and this check wouldn't run